### ISSUE TITLE
Reproducing results from the UMI paper

### ISSUE THREAD

**Issue by Redmar-van-den-Berg** [CGATOxford/UMI-tools_pipelines#4]
I want to replicate the findings from the umi-tools publication, but I cannot find the 'reaper' tool that is mentioned in the requirements.

**Comment by IanSudbery** [comment]
Hi. You can get reapear here: https://www.ebi.ac.uk/research/enright/software/kraken

I'll add it to the requirements documentation.

**Comment by Redmar-van-den-Berg** [comment]
Thanks, I was able to install reaper from the link you posted.

I ran into another issue though, `CGAT` is also listed as a requirement, but the repository appears to be archived, and there is no release `0.2.4`. When I try to run the pipelines I get
```python
Traceback (most recent call last):
  File "../UMI-tools_pipelines/pipeline_scRNASeq.py", line 83, in <module>
    import CGAT.Experiment as E
ImportError: No module named CGAT.Experiment
```

Do you have any suggestions on how to install the correct version of the CGAT module?

**Comment by IanSudbery** [comment]
The best course of action here probably depends on what your aim is. If you are undertaking a reproducibility excercise, then the code for this particular version is here:

https://github.com/CGATOxford/cgat/tree/v0.2.4

You are also going to need from https://github.com/CGATOxford/CGATPipelines (the commit has is in the requirements)

However, I don't know how well the installation procedures will work.

So, if you are just interested in reproducing the results, rather then testing how reproducible the code is, then I recommend you install the modern versions. You will need `cgat-apps` which is installable from `bioconda` and `cgat-flow` which you can get from here:
https://github.com/cgat-developers/cgat-flow

`cgat-flow` has an installation script which will build the a conda envirnoment to run it in. However, I tend to do it manually by cloning the repo and typing

`mamba env create -f conda/environments/cgat-flow-pipelines.yml -p /PATH/TO/LOCATION/FOR/ENV`

activating the environment

`source activate /PATH/TO/LOCATION/FOR/ENV`

then running `python setup.py develop`

If you take this path, the pipelines will need a little light refactoring, but I can probably help with that once exam season is over.

**Comment by Redmar-van-den-Berg** [comment]
The background for my request is that I have developed a reference-free umi deduplication tool, and I would like to compare it's results against UMI-tools. I've been trying to install the original versions in a conda environment, but it appears they are too old to easily install.

Would you recommend I use the latest version of `cgat-apps` and `cgat-flow`, and then run the UMI-tools_pipelines in that environment?

**Comment by IanSudbery** [comment]
Yes, that is probably the best way. Some things have changed in the API since 2015. `CGAT` was split into `cgat` and `cgatcore`. Several of the modules now have lowercase names e.g. `experiment` rather than `Experiment` and `pipeline` rather than `Pipeline`, and most CamelCase names are now underscore_names (e.g. `openFile` is now `open_file`), but its not too difficult a transition and installation is loads easier.

**Comment by Redmar-van-den-Berg** [comment]
@IanSudbery I've been refactoring the pipeline to work with the latest versions of `cgat`, but I'm stuck on report section of the pipeline (`P.run_report` and `P.publish_report`, see [here](https://github.com/Redmar-van-den-Berg/UMI-tools_pipelines/blob/refactor/pipeline_scRNASeq.py#L714)) which do not appear to exist. Has this functionality removed from the latest version? If so, do you have any guidance on what these task should do, and how to replicate this in the latest version?

**Comment by IanSudbery** [comment]
Yes, that functionality had been removed. Those steps can safely be
removed, I believe they are only in the pipeline because they are in the
template pipeline and are not actually used in this instance.

**Comment by Redmar-van-den-Berg** [comment]
@IanSudbery 
I've made a start with refactoring (see [here](https://github.com/Redmar-van-den-Berg/UMI-tools_pipelines/tree/refactor)), but I'm having a hard time installing all required tools in a single conda environment, especially since some of the old cgat code the pipelines rely on is still on python 2. Do you have time to take a look at creating a conda environment that can run these pipelines?
